### PR TITLE
875/slippy map resolver

### LINKED ISSUE #875: New `Driver` to read slippy map tile data
### Kind of request

Changing existing functionality

### Enhancement Description

In https://github.com/Deltares/hydromt/blob/main/hydromt/data_adapter/rasterdataset.py#L360 there is still some custom validation for zoom_level that is not yet present in v1. This can better be moved to a new specific `Driver` or `URIResolver` that specializes in reading slippy map tiles.

In gis/raster.py there is a `to_slippy_tiles` function that results in a DataCatalog entry referring to the previous `raster` driver, which should refer to this new driver.

### Use case

Used if you want to match a certain zoom_level based on your search query.

### PR DESCRIPTION
**Pull request by Jaapel**
## Issue addressed
Fixes #875 
Fixes https://github.com/Deltares/hydromt/issues/984

Also:
- renamed `zoom_level` to either:
  - `zoom` when referring to either an overview level, or a combination of a resolution + unit
  - `overview_level` when only referring to an overview_level

## Explanation
Implemented zoom level in the `ConventionResolver` and the `RasterioDriver`.
I still think we should keep resolvers and driver apart (https://github.com/Deltares/hydromt/issues/1023), then we can implement this much cleaner in the `RasterDataSource`. Now I had to add `zls_dict` to metadata, pass metadata to `URIResolver.resolve` and do a lot of try/excepts to make it happen

## General Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `v1`
- [x] Tests & pre-commit hooks pass
- [x] Updated documentation
- [ ] Updated changelog.rst

### CODE REVIEW

**Comment by savente93** [comment]
does this also resolve #984 ?

**Comment by Jaapel** [comment]
> does this also resolve #984 ?

Yes!

File: hydromt/data_catalog/drivers/raster/rasterio_driver.py [hunk 1]
```diff
@@ -72,24 +75,42 @@ def read_data(
...
+        # Then we can implement looking for a overview level in the driver.
+        def _open() -> Union[xr.DataArray, xr.Dataset]:
+            try:
+                return open_mfraster(uris, **kwargs)
+            except rasterio.errors.RasterioIOError as e:
+                if "Cannot open overview level" in str(e):
+                    kwargs.pop("overview_level")
+                    return open_mfraster(uris, **kwargs)
+                else:
+                    raise
```
Comment (savente93) [inline hydromt/data_catalog/drivers/raster/rasterio_driver.py:111]: should this be `raise e` ?

**Review by savente93 — APPROVED** [review]
can't really attest the logic of the map operations, but the code and tests look good to me, so I think this is fine

File: hydromt/data_catalog/drivers/raster/rasterio_driver.py (hunk 1 above)
Comment (Jaapel) [reply hydromt/data_catalog/drivers/raster/rasterio_driver.py:111]: [You can use both: https://stackoverflow.com/questions/36153805/difference-between-raise-and-raise-e]

Comment (savente93) [reply hydromt/data_catalog/drivers/raster/rasterio_driver.py:111]: cool TIL

**Pull request merged**